### PR TITLE
Add optional parameter to xDSCWebService

### PR DESCRIPTION
**Pull request by mgreenegit**
The new optional parameter will allow for fully automated deployments including SSL when combined with xCertReq.

### CODE REVIEW

**Comment by vors** [comment]
Thank you, Michael.
Please, [fix encoding](https://ci.appveyor.com/project/PowerShell/xwebadministration/build/1.0.8), here are [instructions](https://github.com/PowerShell/DscResources/blob/master/CONTRIBUTING.md#appveyor) about our CI system and common tests.

**Comment by vors** [comment]
@mgreenegit please
1) make [common tests](https://github.com/PowerShell/DscResources/blob/master/CONTRIBUTING.md#common-tests) pass
2) Merge from `dev` branch and resolve merge conflicts
3) Send pull request to **dev** branch.

File: DSCResources/MSFT_xDSCWebService/MSFT_xDSCWebService.psm1 [hunk 1]
```diff
@@ -422,4 +448,22 @@ function Update-LocationTagInApplicationHostConfigForAuthentication
...
 
-Export-ModuleMember -Function *-TargetResource
+# Helper function to filter for valid certificates that match a subject and return the newest valid
+function Find-CertificateThumbprintFromSubject
+{
+    param (
+    [string]$CertificateSubject
+    )
+    
+    $cert = Get-ChildItem Cert:\LocalMachine\my | ? {$_.Subject -eq $CertificateSubject} | ? {$_.NotBefore -lt (get-date) -and $_.NotAfter -gt (get-date)} | Sort-Object NotBefore -Descending | Select -first 1
```
Comment (TravisEz13) [inline DSCResources/MSFT_xDSCWebService/MSFT_xDSCWebService.psm1]: maybe we should throw or give a warning if we find more than one matching cert

**Comment by TravisEz13** [comment]
Closing reoppening to trigger the CLA bot

**Comment by KarolKaczmarek** [comment]
@mgreenegit Can you please resend it to the dev branch?

**Comment by TravisEz13** [comment]
I resolved the merge conflicts and pushed the branch [here](https://github.com/TravisEz13/xPSDesiredStateConfiguration/tree/mgreengit-master), but the [tests failed](https://ci.appveyor.com/project/TravisEz13/xpsdesiredstateconfiguration/build/1.0.2)

**Comment by TravisEz13** [comment]
If there is no update of the pull request within 7 days, we will close the pull request.  After that, if you have an update, feel free to reopen the pull request or a new pull request.

**Comment by TravisEz13** [comment]
Closing because the pull request has not been updated for feedback given for 21 days.  Once you update the code, please feel free to reopen the pull request.  Thanks, Travis.

**Pull request closed without merging**